recipes/libgeotiff/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class LibgeotiffConan(ConanFile):
+    name = "libgeotiff"
+    description = "Libgeotiff is an open source library normally hosted on top " \
+                  "of libtiff for reading, and writing GeoTIFF information tags."
+    license = ["MIT", "BSD-3-Clause"]
+    topics = ("conan", "libgeotiff", "geotiff", "tiff")
+    homepage = "https://github.com/OSGeo/libgeotiff"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        if not self.options.shared or self.options["proj"].shared: # because PROJ is a C++ library with C API
+            del self.settings.compiler.libcxx
+
+    def requirements(self):
+        self.requires.add("libtiff/4.1.0")
+        self.requires.add("proj/6.3.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_UTILITIES"] = False
+        self._cmake.definitions["WITH_TOWGS84"] = True
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=os.path.join(self._source_subfolder, "libgeotiff"))
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "doc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GeoTIFF"
+        self.cpp_info.names["cmake_find_package_multi"] = "GeoTIFF"
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("m")
```

Comment:
```suggestion
            self.cpp_info.system_libs = ["m", "pthread", "dl"]
```

libgeotiff requires all these 3 libs

Reply:
You're right, I also see pthread and dl, but I don't see anything in source code related to dl and pthread.
So, I've compiled libgeotiff and all dependencies of the tree from source code as shared libs. I appears that:
- libtiff also depends on pthread and dl and it's not listed in its recipe.
- xz_utils depends on pthread and it's also not listed.

Reply:
another test:

```bash
conan create -k . libgeotiff/1.5.1@ -o libgeotiff:shared=True -o proj:shared=False -o proj:threadsafe=False -o sqlite3:shared=False -o libtiff:shared=False -o libtiff:jbig=False -o libtiff:jpeg=False -o libtiff:webp=False -o libtiff:lzma=False -o libtiff:zlib=False -o libtiff:zstd=False -b
```

```bash
ldd libgeotiff.so
	linux-vdso.so.1 (0x00007fff2499f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fc2a8a82000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc2a86af000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc2a8311000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc2a80f9000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc2a7d08000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc2a9333000)
```

Reply:
```
conan create -k . libgeotiff/1.5.1@ -o libgeotiff:shared=True -o proj:shared=False -o proj:threadsafe=False -o sqlite3:shared=False -o sqlite3:omit_load_extension=True -o libtiff:shared=False -o libtiff:jbig=False -o libtiff:jpeg=False -o libtiff:webp=False -o libtiff:lzma=False -o libtiff:zlib=False -o libtiff:zstd=False -b
```

```
ldd libgeotiff.so
	linux-vdso.so.1 (0x00007ffe22796000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f6cedb32000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6ced794000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6ced57c000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6ced18b000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6cee5af000)
```

Reply:
nice investigation!